### PR TITLE
Activity Panel: Inbox: Filter update and error notes out of the inbox

### LINKED ISSUE #722: Activity Panel: Inbox: Filter out update and error notices
Right now, the activity panel inbox displays notes of all types. The inbox should NOT be displaying notes of type update or error. Those belong in the separate yet-to-be-built priority notices component instead.

See also #176 (Activity Panel: Inbox) and #115 (Store Alerts aka Priority Notices)

### PR DESCRIPTION
**Pull request by allendav**
Fixes #722

Filters the inbox so that error and update notes do NOT appear. (They belong in the as-yet-not-implemented Store Alerts component.)

### Detailed test instructions:

- Use the https://github.com/Automattic/wc-admin-notes-test-bench plugin to add notes of all four types.
- Navigate to wp-admin > WooCommerce > Dashboard > Inbox
- Make sure the error and update notes do NOT appear
- Make sure the info and warning notes DO appear
- Run the phpunit tests and make sure they pass
- You can run `WC_Tests_API_Admin_Notes::test_get_warning_notes` separately if so desired.
- You can also play with this in Postman:

screen shot 2018-11-19 at 3 56 29 pm

### CODE REVIEW

**Comment by psealock** [comment]
I'm not sure if I have the notes test bench plugin working correctly. Should I be seeing at least one warning message?

screen shot 2018-11-21 at 11 26 02 am

I do see the `type: info,warning` query added to the request, however

File: includes/api/class-wc-admin-rest-admin-notes-controller.php [hunk 1]
```diff
@@ -109,6 +109,12 @@ public function get_items( $request ) {
 			'page'     => $page,
 		);
 
+		$type = isset( $request['type'] ) ? $request['type'] : '';
+		$type = sanitize_text_field( trim( $type ) );
```
Comment (claudiosanches) [inline includes/api/class-wc-admin-rest-admin-notes-controller.php]: `sanitize_text_field()` already trim empty spaces.

```bash
$ wp shell
wp> sanitize_text_field( '  text   ' )
=> string(4) "text"
wp> 
```

**Review by joshuatf — CHANGES_REQUESTED** [review]
@allendav This is testing well for me in the dashboard and only info/warnings are showing up 👍 

The phpunit tests are failing though:
```
1) WC_Tests_API_Admin_Notes::test_get_warning_notes
Failed asserting that 2 matches expected 1.
```
It doesn't look like the types are passed to `get_notes()` so `$escaped_where_types` is empty.  Do we want to set these by default to only show info/warning in the API or change the tests to assert a count of 2?

**Comment by allendav** [comment]
@joshuatf wrote:

> It doesn't look like the types are passed to get_notes() so $escaped_where_types is empty

Thanks for catching that - will fix asap

**Comment by jeffstieler** [comment]
This is ready for another review.

**Comment by timmyc** [comment]
@joshuatf could you give this one another 👀 ?

**Review by joshuatf — APPROVED** [review]

**Pull request merged**